src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -328,31 +317,169 @@ public void Validate(FixedSizeSchemaIndexDef actual)
             }
         }
 
-        public TableSchema CompressValues(FixedSizeSchemaIndexDef etagSource, bool compress)
+        public class DynamicBTreeIndexDef : AbstractBTreeIndexDef
+        {
+            public override TableIndexType Type => TableIndexType.Dynamic;
+
+            public delegate ByteStringContext.Scope IndexValueAction(ByteStringContext context, ref TableValueReader value, out Slice slice);
+
+            public IndexValueAction IndexValueGenerator;
+
+            public override ByteStringContext.Scope GetSlice(ByteStringContext context, ref TableValueReader value,
```

Comment:
Currently `FixedSizeSchemaIndexDef` has `GetValue()`. I think it's more accurate than `GetSlice()` so maybe let's use the same here.

Reply:
done